get_prism_data.R

```diff
@@ -4,39 +4,232 @@ library(prism)
 library(raster)
 library(dplyr)
 library(tidyr)
+library(stringr)
 options(prism.path = "./data/prismdata")
 
-months <- c(1:12)
-clim_vars <- c("ppt", "tmin", "tmean", "tmax")
-for (month in months){
-  for (clim_var in clim_vars){
-    get_prism_monthlys(type=clim_var, year = 1981:2014, month = month, keepZip=F)
+#Years of prism data to 
+yearsToUse=1981:2014
+#Load the DB. create = TRUE does not seem 
+
+sqliteDBFile='./data/bbsforecasting.sqlite'
+if(file.exists(sqliteDBFile)){
+  database <- src_sqlite(sqliteDBFile, create = FALSE)
+} else {
+  database <- src_sqlite(sqliteDBFile, create = TRUE)
+}
+
+
+#######################################################
+#Downloads the raw prism rasters into the folder specified above. 
+#######################################################
+downloadPrism=function(){
+  months <- c(1:12)
+  clim_vars <- c("ppt", "tmin", "tmean", "tmax")
+  for (month in months){
+    for (clim_var in clim_vars){
+      get_prism_monthlys(type=clim_var, year = yearsToUse, month = month, keepZip=F)
+    }
+  }
+}
+
+
+#########################################################
+#Pretty sure postgres isn't used anywhere after adding the raw prism to it,
+#so i'll just comment this out for now. 
+############################################################
+#datadirs = dir(datapath)
+#for (datadir in datadirs) {
+#  bil_file = paste(datadir, '.bil', sep = "")
+#  bil_file_path = file.path(datapath, datadir, bil_file)
+#  sql_file = paste(datadir, ".sql", sep = "")
+#  sql_file_path = file.path(datapath, datadir, sql_file)
+#  system(paste("raster2pgsql -s 4326", bil_file_path, datadir, ">", sql_file_path))
+#  system(paste("psql -d bbsforecasting -f", sql_file_path))
+#}
+
+########################################################
+#Takes output of ls_prism_data() and makes sure all files
+#within a year range are there. ie. for all 12 months and 
+#all 4 variables. 
+#########################################################
+check_If_Prism_Files_Present=function(prismLS, years){
+  #Extract out all the variable names and dates
+  prismLS = prismLS %>%
+    rowwise() %>%
+    mutate(var=strsplit(files, '_')[[1]][2], yearMonth=strsplit(files, '_')[[1]][5]) %>%
+    mutate(year=as.integer(substr(yearMonth,1,4)), month=as.integer(substr(yearMonth, 5,6)), present=1) %>%
+    select(-files, -yearMonth)
+  
+  #Setup a list of what should be there
+  toCheck=expand.grid(year=years, month=1:12, var=c('ppt','tmax','tmean','tmin'))
+  
+  #Left join on the toCheck DF puts present=1 wherever that var/year/month combo was in the prismLS DF.
+  #As long as all of present==1 in the toCheck DF, then all prism rasters are accounted for.
+  toCheck=toCheck %>%
+    left_join(prismLS, by=c('year','month','var'))
+  
+  if(sum(toCheck$present, na.rm=TRUE)==nrow(toCheck)){
+    return(TRUE)
+  } else {
+    return(FALSE)
+  }
+}
```

Comment:
It seems like it would be cleaner to use `identical` to directly compare `prismLS` and `toCheck`. This would save both the merge and the summing and therefore the need to explain how all of this is working, and so could replace lines 66-75 with `return(identical(prismLS, toCheck))`